### PR TITLE
fix: remove tailwind anti-pattern from the repo

### LINKED ISSUE #1522: refactor: remove anti-pattern from the repo
We spent a lot of time rewiring the hot repo after leaning into hardcoding pixels in tailwind. We may want to avoid this anti-pattern 

https://github.com/open-sauced/hot/pull/337

_Originally posted by @bdougie in https://github.com/open-sauced/insights/pull/1500#discussion_r1287280902_

### PR DESCRIPTION
**Pull request by RitaDee**
## Description

This PR includes changes to four different files:

1. `components/atoms/CartIllustration/cart-illustration.tsx`: 
   - Refactored classNames in the className attribute using the nullish coalescing operator (??).
   - Modified h-[34px] and w-[34px] to use h-9 and w-9 respectively.
   - Updated w-[80px] class to w-28.
   - Removed redundant rounded class from a div element.

2. `components/atoms/Checkbox/checkbox.tsx`:
   - Updated the `getId` function to replace all spaces with hyphens and convert the string to lowercase.
   - Updated the `className` attribute of the `CheckboxPrimitive.Root` element to use the `checkboxClasses` variable.
   - Updated the `className` attribute of the `label` element to use the `labelClasses` variable.

3. `components/molecules/RepoRow/repo-row.tsx`:
   - Updated the `w-[55%]` class to `w-5/12`.
   - Updated the `w-[45%]` class to `w-7/12`.

4. `pages/feed/index.tsx`:
   - Updated the `w-[22%]` class to `w-1/5`.
   - Updated the `w-full` class to `w-4/5` in the `DialogContent` element.
   - Updated the `max-w-[48rem]` class to `max-w-screen-xl mx-auto` in the first `div` element.
   - Updated the `hidden w-[30%] xl:w-[24%]` class to `hidden w-1/3 xl:w-1/4` in the second `div` element.

These changes improve the code readability and maintainability.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #1522

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: components/atoms/Checkbox/checkbox.tsx [hunk 1]
```diff
@@ -10,27 +10,29 @@ interface CheckboxProps extends React.ComponentPropsWithoutRef<typeof CheckboxPr
 
 const Checkbox = React.forwardRef<React.ElementRef<typeof CheckboxPrimitive.Root>, CheckboxProps>(
   ({ className, label, id, ...props }, ref) => {
-    const getId = () => (id ? id : label?.replaceAll(" ", "_").toLowerCase());
+    const getId = () => (id ? id : label?.replace(/\s+/g, "-").toLowerCase());
```
Comment (OgDev-01) [inline components/atoms/Checkbox/checkbox.tsx:13]: Why this change?
```suggestion
    const getId = () => (id ? id : label?.replace(" ", "-").toLowerCase());
```

Comment (RitaDee) [reply components/atoms/Checkbox/checkbox.tsx:13]: @OgDev-01 for better readability and alignment of styles.

Comment (RitaDee) [reply components/atoms/Checkbox/checkbox.tsx]: @OgDev-01, I cannot understand why the test is failing.

File: components/atoms/CartIllustration/cart-illustration.tsx [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
   classNames?: string;
 }
+
 const CartIllustration = ({ classNames }: CartIllustrationProps) => {
   return (
     <div
-      className={`inline-flex max-w-[176px] gap-1 items-center px-1.5 w-44 h-12 rounded-md border ${
-        classNames ?? classNames
-      }`}
+      className={`inline-flex max-w-[176px] gap-1 items-center px-1.5 w-44 h-12 rounded-md border ${classNames ?? ""}`}
```
Comment (NsdHSO) [inline components/atoms/CartIllustration/cart-illustration.tsx:8]:
```suggestion
      className={`inline-flex max-w-44 gap-1 items-center px-1.5 w-44 h-12 rounded-md border ${classNames ?? ""}`}
```

File: components/molecules/RepoRow/repo-row.tsx [hunk 3]
```diff
@@ -129,10 +129,10 @@ const RepoRow = ({ repo, topic, userPage, selected, handleOnSelectRepo }: RepoPr
             onCheckedChange={handleSelectCheckbox}
             className={`${user && "border-orange-500 hover:bg-orange-600"}`}
           />
-          <div className="w-[55%]">
+          <div className="w-5/12">
```
Comment (NsdHSO) [inline components/molecules/RepoRow/repo-row.tsx:132]: `7/12` -> `58.33%`

File: components/molecules/RepoRow/repo-row.tsx [hunk 4]
```diff
@@ -129,10 +129,10 @@ const RepoRow = ({ repo, topic, userPage, selected, handleOnSelectRepo }: RepoPr
...
             <TableRepositoryName topic={topic} avatarURL={ownerAvatar} fullName={fullName as string} user={userPage} />
           </div>
-          <div className="w-[45%]">
+          <div className="w-7/12">
```
Comment (NsdHSO) [inline components/molecules/RepoRow/repo-row.tsx:135]: `5/12` -> `41.6%`

File: components/atoms/CartIllustration/cart-illustration.tsx [hunk 5]
```diff
@@ -1,17 +1,14 @@
...
 const CartIllustration = ({ classNames }: CartIllustrationProps) => {
   return (
-    <div
-      className={`inline-flex max-w-[176px] gap-1 items-center px-1.5 w-44 h-12 rounded-md border ${
-        classNames ?? classNames
-      }`}
-    >
-      <div className="h-[34px] w-[34px] border rounded bg-light-slate-5"></div>
-      <div className="w-[80px]  gap-2  rounded">
+    <div className={`inline-flex max-w-44 gap-1 items-center px-1.5 w-44 h-12 rounded-md border ${classNames ?? ""}`}>
```
[the remark this answers is not in the document]
Comment (OgDev-01) [reply components/atoms/CartIllustration/cart-illustration.tsx:7]: For values that don't have a tailwind class equivalent, you can use this [link](https://nekocalc.com/px-to-rem-converter) to convert the px value to rem as a fallback.

File: pages/feed/index.tsx [hunk 6]
```diff
@@ -308,7 +308,7 @@ const Feeds: WithPageLayout<HighlightSSRProps> = (props: HighlightSSRProps) => {
           </TabsList>
 
           {user && (
-            <div className="lg:gap-x-3 px-3 pt-4 flex max-w-[48rem]">
+            <div className="lg:gap-x-3 px-3 pt-4 flex max-w-screen-xl mx-auto">
```
Comment (OgDev-01) [inline pages/feed/index.tsx:311]:
```suggestion
            <div className="lg:gap-x-3 px-3 pt-4 flex max-w-3xl mx-auto">
```

File: pages/feed/index.tsx [hunk 7]
```diff
@@ -324,7 +324,7 @@ const Feeds: WithPageLayout<HighlightSSRProps> = (props: HighlightSSRProps) => {
           <TabsContent value="home">
             <HomeHighlightsWrapper emojis={emojis} mutate={mutate} highlights={data} loading={isLoading} />
             {meta.pageCount > 1 && (
-              <div className="mt-10 max-w-[48rem] flex px-2 items-center justify-between">
+              <div className="mt-10 max-w-screen-lg flex px-2 items-center justify-between">
```
Comment (OgDev-01) [inline pages/feed/index.tsx:327]:
```suggestion
              <div className="mt-10 max-w-3xl flex px-2 items-center justify-between">
```

**Review by OgDev-01 — CHANGES_REQUESTED** [review]

**Comment by OgDev-01** [comment]
If you don't see a matching `%` value in the tailwind classes, ignore updating the percentage to avoid layout breaks 👍.

Nice work so far 🔥🔥

**Comment by RitaDee** [comment]
I will be closing this PR in favour of https://github.com/open-sauced/app/pull/1637.

**Comment by RitaDee** [comment]
I will be closing this PR in favour of https://github.com/open-sauced/app/pull/1637.
